src/PrestaShopBundle/Translation/Provider/ThemeProvider.php

```diff
@@ -24,217 +24,206 @@
-    }
-
-    /**
-     * @param string $themeName The theme name
+     * @param string $locale
+     *
+     * @return MessageCatalogueInterface
      *
-     * @return self
+     * @throws FileNotFoundException
      */
-    public function setThemeName($themeName)
+    public function getFileTranslatedCatalogue(string $locale): MessageCatalogueInterface
     {
-        $this->themeName = $themeName;
+        // load front office catalogue
+        $catalogue = $this->coreFrontProvider->getFileTranslatedCatalogue($locale);
+
+        try {
+            $fileTranslatedCatalogue = (new FileTranslatedCatalogueProvider(
+                $this->getResourceDirectory(),
+                ['*']
+            ))
+                ->getCatalogue($locale);
+
+            // overwrite with the theme's own catalogue
+            $catalogue->addCatalogue($fileTranslatedCatalogue);
+        } catch (FileNotFoundException $e) {
+            // there are no translation files, ignore them
+            return new MessageCatalogue($locale);
+        }
 
-        return $this;
+        return $catalogue;
     }
 
     /**
-     * @param string|null $themeName
+     * @param string $locale
      *
      * @return MessageCatalogueInterface
      */
-    public function getDatabaseCatalogue($themeName = null)
+    public function getUserTranslatedCatalogue(string $locale): MessageCatalogueInterface
     {
-        if (null === $themeName) {
-            $themeName = $this->themeName;
-        }
-
-        return parent::getDatabaseCatalogue($themeName);
+        return (new UserTranslatedCatalogueProvider(
+            $this->databaseReader,
+            ['*'],
+            $this->themeName
+        ))
+            ->getCatalogue($locale);
     }
 
     /**
-     * @throws \Exception
-     *
-     * Will update translations files of the Theme
+     * Check if theme is registered in DB and set class property
      */
-    public function synchronizeTheme()
+    private function validateTheme()
     {
-        $theme = $this->themeRepository->getInstanceByName($this->themeName);
-
-        $path = $this->resourceDirectory . DIRECTORY_SEPARATOR . $this->themeName . DIRECTORY_SEPARATOR . 'translations';
-
-        $this->filesystem->remove($path);
-        $this->filesystem->mkdir($path);
-
-        $this->themeExtractor
-            ->setOutputPath($path)
-            ->setThemeProvider($this)
-            ->extract($theme, $this->locale);
+        $this->theme = $this->themeRepository->getInstanceByName($this->themeName);
 
-        $translationFilesPath = $path . DIRECTORY_SEPARATOR . $this->locale;
-        Flattenizer::flatten($translationFilesPath, $translationFilesPath, $this->locale, false);
-
-        $finder = Finder::create();
-        foreach ($finder->directories()->depth('== 0')->in($translationFilesPath) as $folder) {
-            $this->filesystem->remove($folder);
+        if (!($this->theme instanceof Theme)) {
```

Comment:
I think `$this->themeRepository->getInstanceByName()` will either throws an Exception or returns a `Theme` instance but never something else 🤔
You might want to try catch?